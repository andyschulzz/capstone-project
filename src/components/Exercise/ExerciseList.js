import React from 'react'
import styled from 'styled-components/macro'
import Exercise from './Exercise'
import Button from '../utils/Button'
import { Link } from 'react-router-dom'

export default function ExerciseList({ exerciseData, handleExerciseSelect }) {
  const renderExercises = exerciseData.map((exercise, index) => {
    return (
      <Exercise
        key={exercise.id}
        {...exercise}
        index={index}
        handleExerciseSelect={handleExerciseSelect}
      />
    )
  })

  return (
    <>
      <ButtonWrapper>
        <Button primary>
          <Link to={`/add`}>Add</Link>
        </Button>
      </ButtonWrapper>
      <Wrapper>{renderExercises}</Wrapper>
    </>
  )
}

const ButtonWrapper = styled.div`
  display: flex;
  margin: 20px auto 10px 30px;
  top: 0;

  & a {
    color: white;
    text-decoration: none;
  }
`

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 20px 0 30px 30px;
  gap: 15px;
  overflow-y: auto;
  scroll-behavior: smooth;
`
