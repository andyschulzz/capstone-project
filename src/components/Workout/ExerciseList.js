import React from 'react'
import Exercise from './Exercise'
import * as S from './ExerciseList.styles'
import PropTypes from 'prop-types'

ExerciseList.propTypes = {
  exercises: PropTypes.array.isRequired,
  handleWorkoutAdd: PropTypes.func.isRequired,
}

export default function ExerciseList({ exercises, handleWorkoutAdd }) {
  const lettersAtoZ = [...Array(26)].map((_, i) =>
    String.fromCharCode('A'.charCodeAt(0) + i)
  )

  const renderExercises = lettersAtoZ.reduce((acc, letter) => {
    const letterExercises = filterExercises(letter)
    if (letterExercises.length !== 0) {
      acc.push(<S.Wrapper key={letter}>{letter}</S.Wrapper>)
      letterExercises.forEach(lc => acc.push(lc))
    }
    return acc
  }, [])

  return <>{renderExercises}</>

  function filterExercises(letter) {
    const filter = exercises
      .filter(exercise => String(exercise.name).startsWith(letter))
      .map((exercise, index) => {
        return (
          <Exercise
            key={exercise.id}
            {...exercise}
            handleWorkoutAdd={handleWorkoutAdd}
          />
        )
      })
    return filter
  }
}
