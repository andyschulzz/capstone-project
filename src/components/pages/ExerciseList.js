import React from 'react'
import Exercise from '../Exercise/Exercise.js'
import PageLayout from './PageLayout'
import styled from 'styled-components/macro'

export default function ExerciseList({ exerciseData }) {
  return (
    <PageLayout title="Exercises">
      <ExerciseListStyled>
        {exerciseData.map(exercise => {
          return <Exercise key={exercise.id} {...exercise} />
        })}
      </ExerciseListStyled>
    </PageLayout>
  )
}

const ExerciseListStyled = styled.section`
  display: flex;
  flex-direction: column;
  margin: 20px 0 20px 30px;
  gap: 15px;
  overflow-y: auto;
  scroll-behavior: smooth;
`
