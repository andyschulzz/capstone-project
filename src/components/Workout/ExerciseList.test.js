import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ExerciseList from './ExerciseList.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)
const handleWorkoutExercisesMock = jest.fn()
const exercisesMock = [
  {
    id: '1',
    name: 'Test',
    type: 'Test',
    instructions: 'Test',
    title: 'Test',
  },
]
describe('<ExerciseList />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <ExerciseList
          exercises={exercisesMock}
          handleWorkoutExercises={handleWorkoutExercisesMock}
        />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
