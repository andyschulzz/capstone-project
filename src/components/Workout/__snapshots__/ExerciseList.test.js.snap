// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExerciseList /> should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 0.8rem;
  margin: 20px 0 20px 0;
  color: #888;
  font-weight: 300;
}

<span
    class="c0"
  >
    T
  </span>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 18px 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0 p:first-child {
  margin-top: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 p:first-child {
  margin-top: 0;
}

.c3 {
  font-size: 1rem;
  margin: 10px 0 0 0;
}

.c4 {
  font-size: 0.8rem;
  margin: 0;
  font-weight: 300;
  color: #999;
}

.c1 {
  width: 46px;
  height: 46px;
  margin: 0 36px 0 0;
}

<div
    class="c0"
    data-testid="exercise"
  >
    <img
      alt=""
      class="c1"
    />
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Test
      </p>
      <p
        class="c4"
      >
        Test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
